extensions/interactions/GraphInput/GraphInputRulesServiceSpec.js

```diff
@@ -611,6 +611,499 @@ describe('Graph Input service', function() {
     });
   });
 
+  describe('\'is weakly connected\' rule', function() {
+    it('should return true on empty, complete and cycle graphs', function() {
```

Comment:
I would advise splitting based on directed / undirected graphs rather than true / false answers, since they are very different cases and that will make it easier to see that you have enough checks for both of them.